File: examples/nodejs/subscribe_quote.js
const { Config, QuoteContext, SubType } = require("longport");

let globalCtx;

async function main() {
  let config = Config.fromEnv();
  globalCtx = await QuoteContext.new(config);
  globalCtx.setOnQuote((_, event) => console.log(event.toString()));
  globalCtx.subscribe(["TSLA.US"], [SubType.Quote], true);
}

Promise.all([main()]).catch((err) => console.error(err));

File: examples/python/subscribe_quote.py
from time import sleep
from longport.openapi import QuoteContext, Config, SubType, PushQuote, Period, AdjustType


def on_quote(symbol: str, event: PushQuote):
    print(symbol, event)
    sleep(1)


config = Config.from_env()
ctx = QuoteContext(config)
ctx.set_on_quote(on_quote)
ctx.subscribe(["TSLA.US"], [SubType.Quote], is_first_push=True)
sleep(10000)
